### DISCUSSION TITLE
@mantine/dropzone. Loading state is triggered after files are uploaded. Not before

### DISCUSSION THREAD

**Discussion by Mike-Medvedev** [mantinedev/mantine#6677] [Q&A] [answered]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.12.1

### What package has an issue?

@mantine/dropzone

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

All

### Describe the bug

The loading prop of the dropzone when set to true, creates a loading overlay **after** files have been uploaded, instead of during file upload. I expected to display a spinner as documents upload, not after documents are finished uploading. In the console logs, you can see how loading state triggers after the content script completes the entire file upload before the console.log statement from onDrop() executes. 
![mantine-issue](https://github.com/user-attachments/assets/4bac4d35-5ce0-404d-9f6a-203e85c3421e)
![mantine-issue-example](https://github.com/user-attachments/assets/52677f74-9097-4bb1-afcc-cfda894fadf2)

### If possible, include a link to a codesandbox with a minimal reproduction

https://codesandbox.io/p/sandbox/mantine-react-template-forked-jj6jx9

### Possible fix

Please upload a file and look at the console.log. The fix would be to change where the loading state kicks in. It should be simultaneous with the ondrop starting.

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment, marked as answer]
`Dropzone` component does not handle file uploads – this logic belongs to your application. `loading` prop controls the visibility of the loading overlay, it is changed only by your application as well.
